packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -167,6 +184,18 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ReturnStatement) {
           // handle return statement
 
+          if (options.ignoreVoidInVoid) {
+            if (
+              targetNodeAncestorHasVoidReturnType(invalidAncestor, [
+                AST_NODE_TYPES.FunctionDeclaration,
+                AST_NODE_TYPES.ArrowFunctionExpression,
+                AST_NODE_TYPES.FunctionExpression,
+              ])
```

Comment:
[Refactor] I see no reason why we would want to use `targetNodeAncestorHasVoidReturnType` with some other set of `targets` nodes. So I think we can just put those node types inside the function. WDYT?

Reply:
I agreed that there was no possibility of reuse, so i moved the targets inside the function and changed the function name to be more clear.